clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -555,10 +556,11 @@ public void close() throws IOException {
      * Check if the code is disconnected and unavailable for immediate reconnection (i.e. if it is in
      * reconnect backoff window following the disconnect).
      */
+    @Override
     public boolean isUnavailable(Node node) {
         lock.lock();
         try {
-            return client.connectionFailed(node) && client.connectionDelay(node, time.milliseconds()) > 0;
+            return NetworkClientUtils.isUnavailable(client, node, time);
```

Comment:
Can we move this to a separated PR? This seems to be a refactor.

Reply:
I will do that. Thanks.